bin.js

```diff
@@ -62,3 +55,21 @@ function exec (cmd) {
     stdio: 'inherit'
   })
 }
+
+function buildFromSource () {
+  return hasFlag('--build-from-source')
+}
+
+function verbose () {
+  return hasFlag('--verbose')
+}
+
+function hasFlag (flag) {
+  if (!process.env.npm_config_argv) return false
+
+  try {
+    return JSON.parse(process.env.npm_config_argv).original.indexOf(flag) !== -1
```

Comment:
Note I changed `.cooked` to `.original` since `npm i --verbose` becomes `--loglevel=verbose`. The code is cleaner if we can just check for the original boolean arguments.